src/OpenFin/openfin.ts

```diff
@@ -292,45 +276,42 @@ export class OpenFinContainer extends WebContainerBase {
                 shadow: true
             }
             , () => {
-                const win = contextMenu.getNativeWindow();
+                const win: Window = contextMenu.getNativeWindow();
                 win.document.open("text/html", "replace");
                 win.document.write(OpenFinContainer.menuHtml);
                 win.document.close();
 
-                let menuItemHtml = "";
-                for (const index in menuItems) {
-                    const item = menuItems[index];
-
-                    if (!item.label) {
-                        continue;
-                    }
-
+                let menuItemHtml: string = "";
+                for (const item of menuItems.filter(value => value.label)) {
                     if (!item.id) {
                         item.id = Guid.newGuid();
                     }
 
-                    let imgHtml: string = "<span>&nbsp;</span>";
-                    if (item.icon) {
-                        imgHtml = "<span><img align=\"absmiddle\" class=\"context-menu-image\" src=\"" + this.ensureAbsoluteUrl(item.icon) + "\" /></span>";
-                    }
+                    const imgHtml: string = (item.icon)
```

Comment:
Probably it would worth looking into serving up imgHtml and menuItemHtml from protected methods, therefore those who override menuHtml would be able to override these as well.